src/super_gradients/modules/qarepvgg_block.py

```diff
@@ -316,5 +316,14 @@ def fuse_block_residual_branches(self):
     def from_repvgg(self, src: RepVGGBlock):
         raise NotImplementedError
 
-    def prep_model_for_conversion(self, input_size: Optional[Union[tuple, list]] = None, **kwargs):
-        self.partial_fusion()
+    def prep_model_for_conversion(self, input_size: Optional[Union[tuple, list]] = None, full_fusion: bool = False, **kwargs):
+        """Prepare the QARepVGGBlock for conversion.
+
+        :param full_fusion: If True, performs full fusion, converting the block into a non-trainable, fully fused block.
+                            If False, performs partial fusion, slower for inference but still trainable.
+        """
```

Comment:
I set it to False by default for backward compatibility, but I am not sure if it is relevant.

Reply:
Im pretty sure we need partial fusion for QAT, correct me if im wrong @spsancti .

Reply:
I talked to @spsancti and we chose to set full_fusion as default, since as you said `fuse_repvgg_blocks` is used for QAT and not `prep_model_for_conversion`